spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/MilliSpan.java

```diff
@@ -35,10 +35,10 @@
 	private final long begin;
 	private long end = 0;
 	private final String name;
-	private final String traceId;
+	private final long traceId;
 	@Singular
-	private List<String> parents = new ArrayList<>();
-	private final String spanId;
+	private List<Long> parents = new ArrayList<>();
+	private final Long spanId;
```

Comment:
So you would suggest that it is NOT possible to have nullable span id?

Can you elaborate more on this:

> . Ex. NullTracer also wouldn't need span ids. I understand this is an existing thing, but would like this to be impossible. It is better to make LoggingTracers throw away ids (which itself is odd as most features around logging are in fact correlation!)